en-US/win10/samples/ZigBeeAdapterTutorial.md

```diff
@@ -0,0 +1,203 @@
+
+![ReceiveZdoZcl]({{site.baseurl}}/images/ZigBee/ReceiveZdoZcl.png)
+
+1. Adapter class will build a list of notification it can receive upon initialization. This list contains instance of specific ZigBeeCommand such as DeviceAnnce, ZclReportAttribute.
+2. Upon reception a complete frame from the XBee module, the reception thread of the SerialController will call GetBytesFromModule callback of XBeeModule.
+3. GetBytesFromModule will parse the XBee part of the frame and check if it’s a response to a command that has been sent (see previous section). If not, it will go through the notification list and call the ParseResponse method of each element until 1 accept the frame or none have accepted. In that later case the frame will be thrown away. 
+4. What ParseResponse does is specific to each implementation of the ZigBeeCommand class. For example, the ParseResponse method of the DeviceAnnce class will get the 64 bit address (aka MAC address) and 16 bit address and may be more information about the signaled device and then send “device arrival” signal to BridgeRT (note that several classes are used to achieve that, see code for more). 
+
+### Creating a new ZCL cluster class
+ZigBee adapter doesn’t implement all clusters defined by ZCL. There will be cases where the ZigBee device you want to interact with won’t be supported by ZigBee adapter. In such a case you will need to add support for the missing clusters. 
+
+How to:
+
+1. You need to found out which the ZigBee Profile and ZigBee Device Category each end point of the new device supports. This can be documented by the device manufacturer or discovered by sending some ZDO command to the end points. ZigBee adapter can give you this information if you enable device discovery tracing in Logger class.
+2. Implements the missing clusters (if necessary)
+ - Create a new class that derives from ZclCluster
+ - In constructor add attributes and commands
+3. Update ZigBeeProfileLibrary and ZclClusterFactory accordingly
+4. Don’t forget to update ZclHelper if the cluster you added has either attributes or command that use a not yet supported ZigBee type
+
+Example of a new cluster class
+
+![NewClusterClass]({{site.baseurl}}/images/ZigBee/NewClusterClass.png)
+
+What needs to be done in ZclClusterFactory class
+
+![ZclFactoryChange]({{site.baseurl}}/images/ZigBee/ZclFactoryChange.png)
+
+### Quick overview of frame used by XBee API
+Digi provide API that can be used to interact with its XBee ZigBee module. This API is used to send or receive AT command or ZDO and ZCL command. AT command are specific to XBee module and can only be interpreted by XBee module whereas ZDO or ZCL command can be interpreted by any ZigBee devices. Frame used to send AT command are a bit different than frames used to send ZDO or ZCL command (see XBee ZigBee module documentation for more detail).
```

Comment:
Digi provides an API*
AT commands are specific to the XBee module and can only be interpreted by the XBee module*
Frames used to send AT commands are a bit different than frames used to send ZDO or ZCL commands (see XBee ZigBee module documentation for more details).*